### PR TITLE
Reader: Add My Comments page

### PR DESCRIPTION
**Pull request by stephanethomas**
This pull request seeks to add a new page to the `Reader` that lists comments posted by the user, starting from the most recent ones. It also provides the option to search specific terms:
 
screenshot

This pull request is born from my will to apply the knowledge acquired during the GM class on ElasticSearch. Unfortunately, right now it can only be used by Automatticians (see server patch D7298-code for more details).

#### Testing instructions
 
1. Run `git checkout add/reader-comments` and start your server, or open a [live branch](https://calypso.live/?branch=add/reader-comments)
2. Apply patch D7298-code to your sandbox
3. Point `public-api.wordpress.com` to your sandbox
4. Open the [`Reader` page](http://calypso.localhost:3000/) once logged in
5. Click the `My Comments` link in the sidebar
6. Assert that you are presented with the list of your last comments
7. Enter some search term
8. Assert that you are presented with relevant results
 
#### Reviews
 
- [ ] Code
- [ ] Product
- [ ] Tests

### CODE REVIEW

File: client/reader/comments/comments.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+					line={ translate( 'Comments that you posted will appear here.' ) }
+					illustration={ '/calypso/images/illustrations/illustration-empty-results.svg' }
+					illustrationWidth={ 400 } />
+			);
+		}
+
+		if ( isEmpty( comments ) ) {
+			return (
+				<EmptyContent
+					title={ translate( 'No Comments Found' ) }
```
Comment (a8ci18n) [inline client/reader/comments/comments.js]: Hi! I've found a possible matching string that has already been translated **97** times:
`translate( 'No comments found.' )` *ES Score: 11*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by samouri** [comment]
Thanks for the PR @stephanethomas! I love this idea, its quite neat.

Is this a PR that you are hoping to eventually merge?  If yes then I think we may want to think through the design of it some more and its overlap with the work that @Automattic/lannister is doing on "My Comments" in the calypso admin pages.

GH Project: https://github.com/Automattic/wp-calypso/projects/32
Background: p58i-6u5-p2

**Comment by dmsnell** [comment]
Thanks for this work. It's good stuff to coordinate with Lannister!

Definitely if this is going to go in I think we should sit down and discuss the role the data layer is playing in data fetching and updates. We're trying hard to no longer introduce any `wpcom.js` code or action thunks. There may be more work involved, but there are major benefits to be had as well.

When we were dealing with "grab a possibly massive number of comments" we took an approach different from our norms in building what we called the "comment tree" and the same idea might be useful here as well. Let's talk!

**Comment by gibrown** [comment]
@samouri I chatted briefly with @stephanethomas about this.

I think the way we could integrate this with existing conversations would be to reuse the search piece of this on the A8C Conversations tab. We don't currently have all comments indexed, but this would let us evaluate a search feature on the conversations tab.

For the existing A8C index we could search across all comments, and then match on posts that the user has liked/commented on or also on any comments that the user has made. I don't think we can currently match on comments that were not written by the user unfortunately. But implementing this would be a good way to evaluate whether such a feature would be useful enough in general to put the work and resources into indexing all comments.

It may still require a fair bit of reworking though because I would guess that we'd want to merge the search in with the conversations api so we can have the same api response.

**Comment by stephanethomas** [comment]
Thanks for your feedback everyone! There is no requirement per se to merge this, especially considering that it would only work for Automattician, and that it could be seen as overlapping with the new `Conversations` page. If this can be reused or somehow helpful to others that is great, but my main goal was really to play with ElasticSearch.

**Pull request closed without merging**